### PR TITLE
refactor(storage): add redis structs to storage impls

### PR DESCRIPTION
**Pull request by lsampras**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

- move redis structs to storage implementation crate managing the implementation

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

- splitting implementation of storage from its usage

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/services.rs [hunk 1]
```diff
@@ -179,14 +179,15 @@ impl Store {
         )
         .await;
 
+        #[allow(clippy::expect_used)]
```
Comment (SanchithHegde) [inline crates/router/src/services.rs:182]:
```suggestion
```

File: crates/storage_impl/src/lib.rs [hunk 2]
```diff
@@ -1,2 +1,109 @@
...
 pub mod diesel;
+pub mod redis;
+
+pub use crate::diesel::store::DatabaseStore;
+
+#[allow(dead_code)]
+pub struct RouterStore<T: DatabaseStore> {
+    db_store: T,
+    cache_store: CacheStore,
+    master_encryption_key: Secret<Vec<u8>>,
```
Comment (SanchithHegde) [inline crates/storage_impl/src/lib.rs:14]: Can we change this to a `StrongSecret` instead?

**Review by SanchithHegde — APPROVED** [review]

**Review by ThisIsMani — APPROVED** [review]

File: crates/storage_impl/src/redis.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            .change_context(redis_interface::errors::RedisError::SubscribeError)?;
+
+        // TODO: Handle on message failures
+        // let redis_clone = self.redis_conn.clone();
+        // tokio::spawn(async move {
+        // if let Err(e) = redis_clone.on_message().await {
+        //     logger::error!(pubsub_err=?e);
+        // }
+        // });
+        Ok(())
```
Comment (NishantJoshi00) [inline crates/storage_impl/src/redis.rs:49]: can you add this back?

**Review by NishantJoshi00 — APPROVED** [review]

File: crates/storage_impl/src/redis.rs (hunk 3 above)
Comment (lsampras) [reply crates/storage_impl/src/redis.rs:49]: yeah will add this in the next PR

**Pull request merged**